wvd-templates/DSC/Script-RegisterSessionHost.ps1

```diff
@@ -93,39 +93,55 @@ else {
     Write-Log -Message "Created new Rds RegistrationInfo into variable 'Registered': $obj"
 }
 
-# Executing DeployAgent psl file in rdsh vm and add to hostpool
+# Executing DeployAgent ps1 file in rdsh vm and add to hostpool
 Write-Log "AgentInstaller is $DeployAgentLocation\RDAgentBootLoaderInstall, InfraInstaller is $DeployAgentLocation\RDInfraAgentInstall"
 
 $DAgentInstall = .\DeployAgent.ps1 -AgentBootServiceInstallerFolder "$DeployAgentLocation\RDAgentBootLoaderInstall" `
     -AgentInstallerFolder "$DeployAgentLocation\RDInfraAgentInstall" `
     -RegistrationToken $Registered.Token `
     -StartAgent $true
 
-Write-Log -Message "DeployAgent Script was successfully executed and RDAgentBootLoader,RDAgent installed inside VM for existing hostpool: $HostPoolName`n$DAgentInstall"
+Write-Log -Message "DeployAgent Script was successfully executed and RDAgentBootLoader, RDAgent were installed inside VM for existing hostpool: $HostPoolName`n$DAgentInstall"
 
 # Get Session Host Info
-Write-Log -Message "Getting rdsh host $SessionHostName information"
-
-[PsRdsSessionHost]$pssh = [PsRdsSessionHost]::new("$TenantName", "$HostPoolName", $SessionHostName)
-[Microsoft.RDInfra.RDManagementData.RdMgmtSessionHost]$rdsh = $pssh.GetSessionHost()
-Write-Log -Message "RDSH object content: `n$($rdsh | Out-String)"
+Write-Log -Message "Getting RDSH session host info for '$SessionHostName'"
+
+# Wait for the session host to be available
+$StartTime = Get-Date
+$SessionHost = $null
+$TimeOutInSec = 900
+$DesiredStates = ('Available', 'NeedsAssistance')
+
+Write-Log -Message "Wait for sessions host to be in any of the desired states: $($DesiredStates -join ', ')"
+while ((!$SessionHost -or $SessionHost.Status -notin $DesiredStates) -and (get-date).Subtract($StartTime).TotalSeconds -lt $TimeOutInSec) {
+    write-log "Session host is in '$($SessionHost.Status)' state, continue waiting"

```

Comment:
Maybe output a different message when $SessionHos is null? Accessing $SessionHost.Status should work when it is null but I think will fail if we enable strict mode later.

Reply:
sure